// @tybys/wasm-util/lib/cjs/wasi/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WasiSubclockflags = exports.WasiEventType = exports.WasiFstFlag = exports.WasiClockid = exports.WasiFdFlag = exports.WasiFileControlFlag = exports.FileControlFlag = exports.WasiWhence = exports.WasiRights = exports.WasiFileType = exports.WasiErrno = void 0;
/* eslint-disable spaced-comment */
var WasiErrno;
(function (WasiErrno) {
    WasiErrno[WasiErrno["ESUCCESS"] = 0] = "ESUCCESS";
    WasiErrno[WasiErrno["E2BIG"] = 1] = "E2BIG";
    WasiErrno[WasiErrno["EACCES"] = 2] = "EACCES";
    WasiErrno[WasiErrno["EADDRINUSE"] = 3] = "EADDRINUSE";
    WasiErrno[WasiErrno["EADDRNOTAVAIL"] = 4] = "EADDRNOTAVAIL";
    WasiErrno[WasiErrno["EAFNOSUPPORT"] = 5] = "EAFNOSUPPORT";
    WasiErrno[WasiErrno["EAGAIN"] = 6] = "EAGAIN";
    WasiErrno[WasiErrno["EALREADY"] = 7] = "EALREADY";
    WasiErrno[WasiErrno["EBADF"] = 8] = "EBADF";
    WasiErrno[WasiErrno["EBADMSG"] = 9] = "EBADMSG";
    WasiErrno[WasiErrno["EBUSY"] = 10] = "EBUSY";
    WasiErrno[WasiErrno["ECANCELED"] = 11] = "ECANCELED";
    WasiErrno[WasiErrno["ECHILD"] = 12] = "ECHILD";
    WasiErrno[WasiErrno["ECONNABORTED"] = 13] = "ECONNABORTED";
    WasiErrno[WasiErrno["ECONNREFUSED"] = 14] = "ECONNREFUSED";
    WasiErrno[WasiErrno["ECONNRESET"] = 15] = "ECONNRESET";
    WasiErrno[WasiErrno["EDEADLK"] = 16] = "EDEADLK";
    WasiErrno[WasiErrno["EDESTADDRREQ"] = 17] = "EDESTADDRREQ";
    WasiErrno[WasiErrno["EDOM"] = 18] = "EDOM";
    WasiErrno[WasiErrno["EDQUOT"] = 19] = "EDQUOT";
    WasiErrno[WasiErrno["EEXIST"] = 20] = "EEXIST";
    WasiErrno[WasiErrno["EFAULT"] = 21] = "EFAULT";
    WasiErrno[WasiErrno["EFBIG"] = 22] = "EFBIG";
    WasiErrno[WasiErrno["EHOSTUNREACH"] = 23] = "EHOSTUNREACH";
    WasiErrno[WasiErrno["EIDRM"] = 24] = "EIDRM";
    WasiErrno[WasiErrno["EILSEQ"] = 25] = "EILSEQ";
    WasiErrno[WasiErrno["EINPROGRESS"] = 26] = "EINPROGRESS";
    WasiErrno[WasiErrno["EINTR"] = 27] = "EINTR";
    WasiErrno[WasiErrno["EINVAL"] = 28] = "EINVAL";
    WasiErrno[WasiErrno["EIO"] = 29] = "EIO";
    WasiErrno[WasiErrno["EISCONN"] = 30] = "EISCONN";
    WasiErrno[WasiErrno["EISDIR"] = 31] = "EISDIR";
    WasiErrno[WasiErrno["ELOOP"] = 32] = "ELOOP";
    WasiErrno[WasiErrno["EMFILE"] = 33] = "EMFILE";
    WasiErrno[WasiErrno["EMLINK"] = 34] = "EMLINK";
    WasiErrno[WasiErrno["EMSGSIZE"] = 35] = "EMSGSIZE";
    WasiErrno[WasiErrno["EMULTIHOP"] = 36] = "EMULTIHOP";
    WasiErrno[WasiErrno["ENAMETOOLONG"] = 37] = "ENAMETOOLONG";
    WasiErrno[WasiErrno["ENETDOWN"] = 38] = "ENETDOWN";
    WasiErrno[WasiErrno["ENETRESET"] = 39] = "ENETRESET";
    WasiErrno[WasiErrno["ENETUNREACH"] = 40] = "ENETUNREACH";
    WasiErrno[WasiErrno["ENFILE"] = 41] = "ENFILE";
    WasiErrno[WasiErrno["ENOBUFS"] = 42] = "ENOBUFS";
    WasiErrno[WasiErrno["ENODEV"] = 43] = "ENODEV";
    WasiErrno[WasiErrno["ENOENT"] = 44] = "ENOENT";
    WasiErrno[WasiErrno["ENOEXEC"] = 45] = "ENOEXEC";
    WasiErrno[WasiErrno["ENOLCK"] = 46] = "ENOLCK";
    WasiErrno[WasiErrno["ENOLINK"] = 47] = "ENOLINK";
    WasiErrno[WasiErrno["ENOMEM"] = 48] = "ENOMEM";
    WasiErrno[WasiErrno["ENOMSG"] = 49] = "ENOMSG";
    WasiErrno[WasiErrno["ENOPROTOOPT"] = 50] = "ENOPROTOOPT";
    WasiErrno[WasiErrno["ENOSPC"] = 51] = "ENOSPC";
    WasiErrno[WasiErrno["ENOSYS"] = 52] = "ENOSYS";
    WasiErrno[WasiErrno["ENOTCONN"] = 53] = "ENOTCONN";
    WasiErrno[WasiErrno["ENOTDIR"] = 54] = "ENOTDIR";
    WasiErrno[WasiErrno["ENOTEMPTY"] = 55] = "ENOTEMPTY";
    WasiErrno[WasiErrno["ENOTRECOVERABLE"] = 56] = "ENOTRECOVERABLE";
    WasiErrno[WasiErrno["ENOTSOCK"] = 57] = "ENOTSOCK";
    WasiErrno[WasiErrno["ENOTSUP"] = 58] = "ENOTSUP";
    WasiErrno[WasiErrno["ENOTTY"] = 59] = "ENOTTY";
    WasiErrno[WasiErrno["ENXIO"] = 60] = "ENXIO";
    WasiErrno[WasiErrno["EOVERFLOW"] = 61] = "EOVERFLOW";
    WasiErrno[WasiErrno["EOWNERDEAD"] = 62] = "EOWNERDEAD";
    WasiErrno[WasiErrno["EPERM"] = 63] = "EPERM";
    WasiErrno[WasiErrno["EPIPE"] = 64] = "EPIPE";
    WasiErrno[WasiErrno["EPROTO"] = 65] = "EPROTO";
    WasiErrno[WasiErrno["EPROTONOSUPPORT"] = 66] = "EPROTONOSUPPORT";
    WasiErrno[WasiErrno["EPROTOTYPE"] = 67] = "EPROTOTYPE";
    WasiErrno[WasiErrno["ERANGE"] = 68] = "ERANGE";
    WasiErrno[WasiErrno["EROFS"] = 69] = "EROFS";
    WasiErrno[WasiErrno["ESPIPE"] = 70] = "ESPIPE";
    WasiErrno[WasiErrno["ESRCH"] = 71] = "ESRCH";
    WasiErrno[WasiErrno["ESTALE"] = 72] = "ESTALE";
    WasiErrno[WasiErrno["ETIMEDOUT"] = 73] = "ETIMEDOUT";
    WasiErrno[WasiErrno["ETXTBSY"] = 74] = "ETXTBSY";
    WasiErrno[WasiErrno["EXDEV"] = 75] = "EXDEV";
    WasiErrno[WasiErrno["ENOTCAPABLE"] = 76] = "ENOTCAPABLE";
})(WasiErrno = exports.WasiErrno || (exports.WasiErrno = {}));
var WasiFileType;
(function (WasiFileType) {
    WasiFileType[WasiFileType["UNKNOWN"] = 0] = "UNKNOWN";
    WasiFileType[WasiFileType["BLOCK_DEVICE"] = 1] = "BLOCK_DEVICE";
    WasiFileType[WasiFileType["CHARACTER_DEVICE"] = 2] = "CHARACTER_DEVICE";
    WasiFileType[WasiFileType["DIRECTORY"] = 3] = "DIRECTORY";
    WasiFileType[WasiFileType["REGULAR_FILE"] = 4] = "REGULAR_FILE";
    WasiFileType[WasiFileType["SOCKET_DGRAM"] = 5] = "SOCKET_DGRAM";
    WasiFileType[WasiFileType["SOCKET_STREAM"] = 6] = "SOCKET_STREAM";
    WasiFileType[WasiFileType["SYMBOLIC_LINK"] = 7] = "SYMBOLIC_LINK";
})(WasiFileType = exports.WasiFileType || (exports.WasiFileType = {}));
const FD_DATASYNC = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(0));
const FD_READ = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(1));
const FD_SEEK = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(2));
const FD_FDSTAT_SET_FLAGS = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(3));
const FD_SYNC = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(4));
const FD_TELL = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(5));
const FD_WRITE = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(6));
const FD_ADVISE = ( /*#__PURE__*/BigInt(1) << /*#__PURE__*/ BigInt(7));
const FD_ALLOCATE = ( /*#__PURE__*/BigInt(1) << /
